File: 1.TwoSum.java
class Solution {
	public int[] twoSum(int[] nums, int target) {
		int[] result = new int[2];

		if (nums == null || nums.length == 0) {
			return result;
		}

		HashMap<Integer, Integer> existed = new HashMap<>();

		for (int i = 0; i < nums.length; i++) {
			if (existed.containsKey(target - nums[i])) {
				result[0] = existed.get(target - nums[i]);
				result[1] = i;
				return result;
			}
			existed.put(nums[i], i);
		}

		return result;
	}
}

File: 1.TwoSum.js
var twoSum = function (nums, target) {
  var map = {};
  for (var i = 0; i < nums.length; i++) {
    if (map[nums[i]] !== undefined) {
      return [parseInt(map[nums[i]]), parseInt(i)];
    } else {
      map[target - nums[i]] = i;
    }
  }
};